flamedisx/xenon/x1t_sr1.py

```diff
@@ -207,6 +214,13 @@ def add_extra_columns(self, d):
                           d['y'].values,
                           d['z'].values]))
 
+        # Not too good. patchy. event_time should be int since event_time in actual
+        # data is int64 in ns. But need this to be float32 to interpolate.
+        if 'elife' not in d.columns:
+            d['event_time'] = d['event_time'].astype('float32')
```

Comment:
Maybe double precision would be necessary here since we deal with interpolations of ns

Reply:
Good suggestion. I gave it some thoughts and realised that keeping it as float32 might be better. This is vaguely related to https://github.com/FlamTeam/flamedisx/issues/90 about choosing between int and float32, but the option of using float64 instead float32 does not seem to present much advantages.

The precision is constrained by the elife map from MongoDB. The timestamps are the UNIX time in s, not ns, even though they are stored as float64.

In this PR, the elife maps are read in using `read_maps_tf` as `fd.float_type()` (see https://github.com/FlamTeam/flamedisx/blob/b9c98fbae154a055f3527d017bd05c8a9aa8b1ab/flamedisx/xenon/x1t_sr1.py#L89), where `fd.float_type` is defined as `tf.float32` (see https://github.com/FlamTeam/flamedisx/blob/a338ab35b9ce0fc1649620dd3de113021468caa8/flamedisx/utils.py#L41), which is why `d['event_time']` is required to be float32.

As mentioned by Bart in https://github.com/FlamTeam/flamedisx/issues/90, most of the internal computations are in float32, though there are some exceptions, especially in `likelihood.py`.

Reply:
Yeah I agree. The thing that made me a bit worry is the fact that you already lose some info going from int64 to float64 when dealing with large numbers (millenium bug, somehow related). In float32 you definitely loose nanosecond precision if data is in int64(ns). I guess the comment can be misleading. Anyway in case we ever need a proper interpolation and we see weird things this is a thing to be aware of.

Reply:
ok. i'll just reference https://github.com/FlamTeam/flamedisx/issues/90 in the comments instead.